File: android/app/src/main/java/com/reactnativestarter/ExampleNativeModule.java
package com.reactnativestarter;

import android.util.Log;

import com.facebook.react.bridge.Promise;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;

public class ExampleNativeModule extends ReactContextBaseJavaModule {

    private static final String MODULE_NAME = "ExampleNativeModule";

    public ExampleNativeModule(ReactApplicationContext reactContext) {
        super(reactContext);
    }

    // This is how RN bridge knows about our module
    @Override
    public String getName() {
        return MODULE_NAME;
    }

    /**
     * Method called from RN app
     * Example method that reverses character order
     * @param text (string)
     */
    @ReactMethod
    public void reverseCharacterOrder(String text, Promise promise) {
        StringBuilder reversedString = new StringBuilder();
        reversedString.append(text);
        reversedString = reversedString.reverse();
        promise.resolve(reversedString.toString());
    }
}

File: ios/ExampleNativeModule.m
//  ExampleNativeModule.m
#import "ExampleNativeModule.h"

@implementation ExampleNativeModule

// This is how RN bridge knows about our module
RCT_EXPORT_MODULE();

// Method called from RN app
// Example method that reverses character order
RCT_EXPORT_METHOD(reverseCharacterOrder:(NSString *)text resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject) {
  NSMutableString *reversedString = [NSMutableString string];
  NSInteger charIndex = [text length];
  
  while (text && charIndex > 0) {
    charIndex--;
    NSRange subStrRange = NSMakeRange(charIndex, 1);
    [reversedString appendString:[text substringWithRange:subStrRange]];
  }
  
  resolve(reversedString);
}

@end

